TACC/Frontera-Portal
# task/FP-503 - Onboarding -- refactor system creation step

## Overview: ##

WIP: Moving system creation from UserSystemsManager to a new systems_creation step

## PR Status: ##


## Related Jira tickets: ##

* [FP-503](https://jira.tacc.utexas.edu/browse/FP-503)
* [FP-669](https://jira.tacc.utexas.edu/browse/FP-669)

## Summary of Changes: ##

- Move PORTAL_DATA_DEPOT_LOCAL_STORAGE_SYSTEMS from settings_secret to settings
- Add named templates for PORTAL_DATA_DEPOT_LOCAL_STORAGE_SYSTEMS
- Refactor UserSystemsManager get_system_id and get_sys_tas_user_dir to use named templates
- Alter jobs post so that it uses UserSystemsManager to compute default archive system name
- Refactor AllocationStep to perform synchronous, forced retrieval of TAS allocations and subsequent caching in ElasticSearch
- Set AllocationStep as a default onboarding step setting

## Testing Steps: ##
1. Delete current user from local testing environment
2. On first login, you will be directed to the temporary onboarding page (which has no data.) Check your logs to see that it has retrieved your TAS allocations, or perform the following in the python shell to see the events:

```
from portal.apps.onboarding.models import SetupEvent

SetupEvent.objects.all()
```

If you try to navigate to the workbench a second time, you should make it there (since the allocation retrieval will have been completed)

3. Make sure you still have access to frontera and longhorn homes
4. Make sure default archive location on job submission archives to frontera


## UI Photos:

## Notes: ##

Unit tests currently fail due to missing index.html. This needs to be corrected in pipeline - the file needs to be rendered from jinja template.
